# --- SDE-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
#
# Filename: package/.../librsync/03-getopt.patch
# Copyright (C) 2011 The OpenSDE Project
#
# More information can be found in the files COPYING and README.
#
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- SDE-COPYRIGHT-NOTE-END ---

Author: Daniel Baumann <daniel@debian.org>
Description:
 Makes rdiff aware of the -i and -z shortcuts for --gzip and --bzip2 options
 (Closes: #435894).

diff -Naurp librsync.orig/rdiff.c librsync/rdiff.c
--- librsync.orig/rdiff.c	2008-07-01 15:29:15.000000000 +0200
+++ librsync/rdiff.c	2008-07-01 15:25:17.000000000 +0200
@@ -97,8 +97,8 @@ const struct poptOption opts[] = {
     { "sum-size",    'S', POPT_ARG_INT,  &strong_len },
     { "statistics",  's', POPT_ARG_NONE, &show_stats },
     { "stats",        0,  POPT_ARG_NONE, &show_stats },
-    { "gzip",         0,  POPT_ARG_NONE, 0,             OPT_GZIP },
-    { "bzip2",        0,  POPT_ARG_NONE, 0,             OPT_BZIP2 },
+    { "gzip",        'z', POPT_ARG_NONE, 0,             OPT_GZIP },
+    { "bzip2",       'i', POPT_ARG_NONE, 0,             OPT_BZIP2 },
     { "paranoia",     0,  POPT_ARG_NONE, &rs_roll_paranoia },
     { 0 }
 };
